### PR TITLE
Add utility scripts for Kubeflow: Machine Learning with Kubernetes

### PR DESCRIPTION
**Pull request by Rishit-dagli**
This is the first blog in the Kubeflow series.

Developing and deploying machine learning systems could be a pain with multiple things you need to manage. In this article, I introduce you and help you get started with Kubeflow while also understanding how Kubeflow works. This is the first article in the Kubeflow series and I will try to help you answer the question "Why and When Kubeflow?" and understand the Kubeflow architecture coupled with some of my experiences and tips.

### CODE REVIEW

**Comment by saiyam1814** [comment]
@pavangudiwada can you review this ?

**Comment by saiyam1814** [comment]
@pavangudiwada all looking good ?

File: _posts/2022-07-11-kubeflow-machine-learning-on-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+- Multi Tenancy: Kubeflow is built in a manner to allow multiple people to easily collaborate on a project. Multi tenancy is built around user namespaces and user access is defined through RBAC (role-based access control) policies which already exist in Kubernetes. You could easily restrict each user to only worry about and interact with the Kubeflow components in their configuration.
```
Comment (pavangudiwada) [inline _posts/2022-07-11-kubeflow-machine-learning-on-kubernetes.md]: Would be great if this part is spaced like this -> 
image

and side headings like "Training Operators:" be mentioned using "**Training Operators**" will help with better redability.

File: _posts/2022-07-11-kubeflow-machine-learning-on-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+6. Deploy the model: Finally, now you can deploy your model which among others requires you to serve your model over a REST API, maintain model versions, perform model rollbacks, canary deployments, batch prediction, and scaling. You can use *KServe* or also use *Kubeflow pipelines* to orchestrate other ways of serving your models like TensorFlow Extended, TorchServe, TensorRT, and more.
+
+7. Maintenance: The process, unfortunately, does not end at the "Deploy" step; you also need to be able to monitor live data and model prediction distributions and monitor for data and concept drifts. This data could also tell you if you need to retrain your model or what specific changes you might need to make. This stage is a valuable part of any ML workflow and with Kubeflow you could also use integrate *TensorBoard* to do just this.
```
Comment (pavangudiwada) [inline _posts/2022-07-11-kubeflow-machine-learning-on-kubernetes.md]: Side headings with **Bold** just like the previous one.

**Comment by pavangudiwada** [comment]
> @pavangudiwada all looking good ?

Just some minor suggestions. Everything else looks very good.

**Comment by Rishit-dagli** [comment]
@pavangudiwada 
Thanks for the reviews, I have fixed those. I am syncing this Markdown document with Hashnode as well so that should be taken care of.

Some questions before we merge this:

- Will we be hosting this backup on something like GitHub pages?
- Before merging, we also need to add a canonical URL to the version of this blog on https://kubesimplify.com/ since this is just a backup and out there for simplifying reviews, right?
- Were you able to see the Hashnode rendering too? TL: DR It renders properly however do the references look well rendered to you?
- Any ideas on an interesting cover image or title?
- Do you think it is a nice idea to add something like that?
```docs
We will take forward what we talk about in this article in the article in this series where we will take a deeper dive into Kubeflow Notebooks, until then, adieu.
```

**Comment by Rishit-dagli** [comment]
Here's a preview for this PR: https://kubeflow-series-preview.netlify.app/ . Maybe later we can set it up to run on PRs iff we are keen on that since this is just a backup and a place for reviews anyways and our main source is kubesimplify.com

**Comment by Rishit-dagli** [comment]
Since the article is already out now, I'll merge this

**Pull request merged**